frontend/lib/identification/animated_progressbar.dart

```diff
@@ -0,0 +1,52 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+
+class AnimatedProgressbar extends StatefulWidget {
+  final int duration;
+  final VoidCallback onCompleted;
+
+  const AnimatedProgressbar({Key key, this.duration, this.onCompleted})
+      : super(key: key);
+
+  @override
+  _AnimatedProgressbarState createState() => _AnimatedProgressbarState();
+}
+
+class _AnimatedProgressbarState extends State<AnimatedProgressbar>
+    with SingleTickerProviderStateMixin {
+  AnimationController controller;
+  Animation<double> animation;
+
+  @override
+  void initState() {
+    super.initState();
+    controller = AnimationController(
+        duration: Duration(
+            milliseconds: widget.duration > 0 ? widget.duration : 1 << 63),
+        vsync: this);
+    animation = Tween(begin: 1.0, end: 0.0).animate(controller)
+      ..addListener(() {
+        setState(() {});
+      })
```

Comment:
What does that do?

Reply:
It updates the widget when the progress value changes